### PR TITLE
Flow policy experimental docs

### PR DESCRIPTION
**Pull request by tmbo**
**Proposed changes**:
- Flow policy experimental docs

fixes https://rasahq.atlassian.net/browse/ENG-356

### CODE REVIEW

File: docs/docs/llms/flow-policy.mdx [hunk 1]
```diff
@@ -3,9 +3,179 @@ id: flow-policy
...
+progresses based on user input and system response. Other classifiers will
+modify the underlying state machine to handle edge-cases. 
+
+Please refer to the [flows specification](../flows.mdx) for detailed 
+information on defining flows.
+
+## Adding the FlowPolicy to your bot
+
+Incorporating the FlowPolicy into your bot involves a two-step process. First,
+you need to configure the FlowPolicy in your 'config.yml' file. The
```
Comment (m-vdb) [inline docs/docs/llms/flow-policy.mdx:40]:
```suggestion
you need to configure the `FlowPolicy` in your `config.yml` file. The
```

File: docs/docs/llms/flow-policy.mdx [hunk 2]
```diff
@@ -3,9 +3,179 @@ id: flow-policy
...
 
-<RasaDiscoveryBanner/>
+<RasaDiscoveryBanner />
+
+The Flow Policy in Rasa is a state machine that enables you to
+manage conversational flows in your chatbot and combine your business logic 
+with machine learning. 
+
+The policy manages the state of your bot, transitioning between states, and
+initiating new flows when required. The "Flow Stack" represents
```
Comment (m-vdb) [inline docs/docs/llms/flow-policy.mdx:20]: Do we want to explain that the flow stack is LIFO? I feel like most developers know what a stack is, but not assuming prior knowledge is one of our documentation principles. Maybe something for later though. Up to you

File: docs/docs/llms/flow-policy.mdx [hunk 3]
```diff
@@ -3,9 +3,179 @@ id: flow-policy
...
+/>
+
+The correction *"oh sorry I meant John"* from the user triggers a correction
+flow to be started. This **correction is not explicitly modeled** in the 
+`transfer_money` flow but rather handled by a so called 
+"conversational pattern".
+
+The correction flow is one of these "conversational patterns". It is only 
+executed when the user corrects a prior input. When triggered, the 
+correction flow is put on top of the flow stack and executed.
```
Comment (m-vdb) [inline docs/docs/llms/flow-policy.mdx:102]:
```suggestion
The correction flow is only  executed when the user wants to correct a prior input.
When triggered, the correction flow is put on top of the flow stack and executed. 
```

File: docs/docs/llms/flow-policy.mdx [hunk 4]
```diff
@@ -3,9 +3,179 @@ id: flow-policy
...
+flow to be started. This **correction is not explicitly modeled** in the 
+`transfer_money` flow but rather handled by a so called 
+"conversational pattern".
+
+The correction flow is one of these "conversational patterns". It is only 
+executed when the user corrects a prior input. When triggered, the 
+correction flow is put on top of the flow stack and executed. 
+
+The correction flow is an "interruption" to the `transfer_money` flow. Once the
+correction flow is completed, the `transfer_money` flow is resumed. Before, the
```
Comment (m-vdb) [inline docs/docs/llms/flow-policy.mdx:105]:
```suggestion
correction flow is completed, the `transfer_money` flow is resumed. Before the
```

File: docs/docs/llms/flow-policy.mdx [hunk 5]
```diff
@@ -3,9 +3,179 @@ id: flow-policy
...
+`transfer_money` is resumed, the `resume_flow` flow is triggered. It allows the
+dialogue designer to define how to guide the user back to the original flow,
```
Comment (m-vdb) [inline docs/docs/llms/flow-policy.mdx:107]: I'm not sure how it technically allows the dialogue designer to define how to guide the user back to the orginal flow. Can you detail?

File: docs/docs/llms/flow-policy.mdx [hunk 6]
```diff
@@ -3,9 +3,179 @@ id: flow-policy
...
+
+The FlowPolicy transitions between states in response to user inputs or after an
+action has been completed. Initially, it waits for user input. Once received, it
+advances the state of the machine until it makes an action prediction. After the
+action has been completed and its result incorporated, it continues advancing to
+the next state.
+
+### Starting New Flows
+
+Starting a new flow in the FlowPolicy can be done in one of two ways. Either,
```
Comment (m-vdb) [inline docs/docs/llms/flow-policy.mdx:125]:
```suggestion
Starting a new flow in the `FlowPolicy` can be done in one of two ways. Either,
```

File: docs/docs/llms/flow-policy.mdx (hunk 2 above)
Comment (tmbo) [reply docs/docs/llms/flow-policy.mdx:20]: will do so further down :+1:

File: docs/docs/llms/flow-policy.mdx [hunk 7]
```diff
@@ -3,9 +3,179 @@ id: flow-policy
...
+        action: utter_lets_transfer_money
+    # - ... 
+```
+
+The NLU model will predict the intent `transfer_money` and the FlowPolicy will
+start the flow `transfer_money` as a result.
+
+#### Predicting Flow Actions
+Another policy can predict the action `flow_transfer_money` and the FlowPolicy
+will start the flow `transfer_money` as a result.
```
[the remark this answers is not in the document]
Comment (tmbo) [reply docs/docs/llms/flow-policy.mdx:150]: added an faq :+1:

**Pull request merged**